src/javafundamentals/classes/BananaBasket.java

```diff
@@ -0,0 +1,25 @@
+package javafundamentals.classes;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class BananaBasket {
+    private List<Banana> bananaList = new ArrayList<>();
+    private int noOfBananas;
+
+    public void addBananas (Banana ...bananas) {
+        bananaList.addAll(Arrays.asList(bananas));
+        noOfBananas = bananaList.size();
+    }
+
+    public Banana takeBanana (){
+        Banana returnBanana = bananaList.remove(bananaList.size()-1);
+        noOfBananas = bananaList.size();
+        return returnBanana;
+    }
+
+    public int getNoOfBananas() {
+        return noOfBananas;
```

Comment:
If you return here `bananaList.size()` then you dont need the variable `noOfBananas` .
Normally this auxiliar variable is when you implement this with an `Array`

Reply:
Okay good to know